osquery/tables/system/darwin/preferences.cpp

```diff
@@ -0,0 +1,179 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <CoreFoundation/CoreFoundation.h>
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+
+namespace osquery {
+namespace tables {
+
+// Walk the supported preference value structures.
+void genOSXPreferenceValues(const CFTypeRef& value,
+                            const Row& base,
+                            QueryData& results);
+
+void genOSXHashPreferences(const CFDictionaryRef& hash,
+                           const Row& base,
+                           QueryData& results) {
+  CFIndex count = CFDictionaryGetCount(hash);
+
+  // Allocate memory for copies of each key/value in dictionary.
+  auto keys = (void**)malloc(sizeof(void*) * count);
+  auto values = (void**)malloc(sizeof(void*) * count);
```

Comment:
Unfortunately, not as far as I am aware, https://developer.apple.com/library/ios/documentation/CoreFoundation/Reference/CFDictionaryRef/index.html#//apple_ref/c/func/CFDictionaryGetKeysAndValues